synapse/storage/databases/main/room.py

```diff
@@ -1411,6 +1411,54 @@ async def add_event_report(
             desc="add_event_report",
         )
 
+    async def get_event_report(self, report_id: int) -> Dict[str, Any]:
+        """Retrieve a an event report
+
+        Args:
+            report_id: ID of reported event in database
+        Returns:
+            event_report: json list of information from event report
+        """
+
+        def _get_event_report_txn(txn, report_id):
+
+            sql = """
+                SELECT
+                    er.id,
+                    er.received_ts,
+                    er.room_id,
+                    er.event_id,
+                    er.user_id,
+                    er.reason,
+                    er.content,
+                    events.sender,
+                    room_aliases.room_alias,
+                    event_json.json AS event_json
+                FROM event_reports AS er
+                LEFT JOIN room_aliases
+                    ON room_aliases.room_id = er.room_id
+                JOIN events
+                    ON events.event_id = er.event_id
+                JOIN event_json
+                    ON event_json.event_id = er.event_id
+                WHERE er.id = ?
+            """
+
+            txn.execute(sql, [report_id])
+            event_report = self.db_pool.cursor_to_dict(txn)
+
+            try:
+                event_report["content"] = db_to_json(event_report["content"])
+                event_report["event_json"] = db_to_json(event_report["event_json"])
+            except Exception:
+                pass
```

Comment:
I'm wary that we missed the above bug because this try/except clause was so broad. Perhaps we can test for issues in the data more thoroughly?

Reply:
I am not sure which issues can happen. It is not used consistently at the same way.

https://github.com/matrix-org/synapse/blob/8a4a4186ded34bab1ffb4ee1cebcb476890da207/synapse/storage/databases/main/search.py#L156-L160

https://github.com/matrix-org/synapse/blob/0a08cd10656349fc9e59d90c1cb3170598b7b0da/synapse/storage/databases/main/events_worker.py#L651-L655

Reply:
Yeah, after taking a look it looks like this function can return `JSONDecodeError`, `UnicodeDecodeErrors`, etc.

I think it's probably best to just leave it as `Exception` then.